hudi-flink-datasource/hudi-flink/src/test/java/org/apache/hudi/table/ITTestHoodieDataSource.java

```diff
@@ -781,6 +781,34 @@ void testLookupJoin(HoodieTableType tableType) {
     assertRowsEquals(result, TestData.DATA_SET_SOURCE_INSERT);
   }
 
+  @Test
+  void testDeleteForLegacyAvroWrite() {
```

Comment:
It makes more sense we add tests in `TestInputFormat`  for payload merging logic in the future.